<launch>
	<arg name="doDebug" default="false" />
	<arg if="$(arg doDebug)" name="launchPrefix" value="xterm -e gdb --args" />
	<arg unless="$(arg doDebug)" name="launchPrefix" value="" />
	<arg name="camera_frame_id" default="raspicam"/>
	<arg name="camera_id" default="0"/>
	<arg name="camera_name" default="camerav2_1280x720"/>
	<arg name="enable_imv" default="false"/>
	<arg name="enable_raw" default="false"/>

	<!-- top - 00:09:01 up  4:12,  3 users,  load average: 4.76, 1.94, 0.97 w 90fps -->
	
	<node 
		type="raspicam_node" 
		pkg="raspicam_node" 
		name="raspicam_node" 
		output="screen"
		launch-prefix="$(arg launchPrefix)" >
		<param name="camera_frame_id" value="$(arg camera_frame_id)"/>  
		<param name="camera_id" value="$(arg camera_id)"/> 
		<param name="camera_info_url" value="package:://wr_aruco/cfg/ost.yml" />
		<!--param name="camera_info_url" value="package://pibota/camera_info/camerav2_1280x720.miranda.yaml"/-->
		<param name="camera_name" value="$(arg camera_name)"/>
	    <param name="enable_imv" value="$(arg enable_imv)"/>
		<param name="enable_raw" value="$(arg enable_raw)"/>
		<!-- We are running at 90fps to reduce motion blur -->
		<param name="framerate" value="25"/>
		<param name="height" value="720"/>
		<param name="hFlip" value="true" />
		<param name="vFlip" value="true" />
		<param name="width" value="1280"/>
	</node>

	<node 
		type="republish" 
		pkg="image_transport" 
		name="compressed_to_raw_video" 
		output="screen"
		args="compressed in:=/raspicam_node/image out:=/raspicam_node/raw_image" >
	</node>

	<!--node
		type="camera_calibration"
		pkg="cameracheck.py"
		name="camera_calibration"
		output="screen"
		args="-s 8x6 -q=0.034"
		>
		<remap from="image" to="/raspicam_node/raw_image" />
	</node-->

</launch>
